### PR TITLE
fix: Forking fails when template has REST datasource with Bearer Token Authentication

### LINKED ISSUE #19415: [Bug]: Import application flow for template application with Rest API datasources requests datasource reconnection
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

To avoid reauthenticating/reconnecting the datasources which are the part of template application, we are exporting the application with datasource credentials. But for the rest API datasources where we have `ApiKeyAuth` and `BearerTokenAuth` as authentication types, the import flow fails to authenticate the datasource.

### Steps To Reproduce

1. Create new template application. (`exportWithConfiguration: true`)
2. Add rest API datasource and add the auth of type `ApiKeyAuth/BearerTokenAuth`. You can use any free service like clearbit for this.
3. Export the application
4. Import the application, you will be asked to reconnect the datasource again. 

### Public Sample App

_No response_

### Issue video log

_No response_

### Version

Cloud: v1.8.14

### PR DESCRIPTION
**Pull request by nayan-rafiq**
## Description
When an application is exported with credentials i.e. template application, it does not contain the bearer token authentication credentials. As a result, when user is trying to import this application, the datasource configuration popup appears. This PR fixes this issue.

TL;DR enable export-import application with bearer token authentication

Fixes #19415 

Media
> A video or a GIF is preferred. when using Loom, don’t embed because it looks like it’s a GIF. instead, just link to the video

## Type of change
- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Manual
- Jest

### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

### QA activity:
- [ ] Test plan has been approved by relevant developers
- [ ] Test plan has been peer reviewed by QA
- [ ] Cypress test cases have been added and approved by either SDET or manual QA
- [ ] Organized project review call with relevant stakeholders after Round 1/2 of QA
- [ ] Added Test Plan Approved label after reveiwing all Cypress test

### CODE REVIEW

**Comment by abhvsn** [comment]
@sidhantgoel @AnaghHegde can you guys please take a look at this PR little occupied with other tasks.

**Review by sidhantgoel — APPROVED** [review]
Looks good
